Content.IntegrationTests/DMProject/code.dm

```diff
@@ -1,274 +1,50 @@
-	for(,j <= 3,j++)
-		counter++
-	. += counter
-	counter = 0
-
-	j = 1
-	for(,j++ <= 3)
-		counter++
-	. += counter
-
-/matrix/proc/Equals(matrix/mat)
-	return src.a == mat.a && src.b == mat.b && src.c == mat.c && src.d == mat.d && src.e == mat.e && src.f == mat.f
-
-/world/proc/test_matrix_add()
-	var/matrix/M = matrix(1, 2, 3, 4, 5, 6)
-	var/matrix/N = matrix(7, 8, 9, 10, 11, 12)
-
-	M.Add(N)
-
-	if(!M.Equals(matrix(8, 10, 12, 14, 16, 18)))
-		CRASH("Unexpected matrix/Add result")
-
-/world/proc/test_matrix_invert()
-	var/matrix/M = matrix(1, 2, 3, 4, 5, 6)
-
-	M.Invert()
-
-	if(!M.Equals(matrix(-1.6666667, 0.6666667, 1, 1.3333334, -0.33333334, -2)))
-		CRASH("Unexpected matrix/Invert result [M.a] [M.b] [M.c] [M.d] [M.e] [M.f]")
-
-	M.Invert()
-
-	if(!M.Equals(matrix(1, 2, 3, 4, 5, 6)))
-		CRASH("Unexpected matrix/Invert result2 [M.a] [M.b] [M.c] [M.d] [M.e] [M.f]")
-
-/world/proc/test_matrix_multiply()
-	var/matrix/M = matrix(1, 2, 3, 4, 5, 6)
-	var/matrix/N = matrix(7, 8, 9, 10, 11, 12)
-
-	M.Multiply(N)
-
-	if(!M.Equals(matrix(39, 54, 78, 54, 75, 108)))
-		CRASH("Unexpected matrix/Multiply result")
-
-/world/proc/test_matrix_scale()
-	var/matrix/M = matrix(1, 2, 3, 4, 5, 6)
-
-	M.Scale(2)
-
-	if(!M.Equals(matrix(2, 4, 6, 8, 10, 12)))
-		CRASH("Unexpected matrix/Scale result")
-
-/world/proc/test_matrix_subtract()
-	var/matrix/M = matrix(1, 2, 3, 4, 5, 6)
-	var/matrix/N = matrix(7, 8, 9, 10, 11, 12)
-
-	M.Subtract(N)
-
-	if(!M.Equals(matrix(-6, -6, -6, -6, -6, -6)))
-		CRASH("Unexpected matrix/Subtract result")
-
-/world/proc/test_matrix_translate()
-	var/matrix/M = matrix(1, 2, 3, 4, 5, 6)
-
-	M.Translate(2)
-
-	if(!M.Equals(matrix(1, 2, 5, 4, 5, 8)))
-		CRASH("Unexpected matrix/Translate result")
-
-/world/proc/test_matrix_turn()
-	var/matrix/M = matrix(1, 2, 3, 4, 5, 6)
-
-	M.Turn(90)
-
-	if(!M.Equals(matrix(4, 5, 6, -1, -2, -3)))
-		CRASH("Unexpected matrix/Turn result")
-
-/world/proc/matrix_operations_test()
-	test_matrix_add()
-	// Invert currently fails because OpenDream doesn't handle floats the way BYOND does.
-	// Uncomment when that's changed.
-	// test_matrix_invert()
-	test_matrix_multiply()
-	test_matrix_scale()
-	test_matrix_subtract()
-	test_matrix_translate()
-	test_matrix_turn()
-
-/world/proc/unicode_procs_test()
-	ASSERT(length("😀") == 2)
-	ASSERT(length_char("😀") == 1)
-
-	// This is the combination of the Man, Woman, Girl and Boy emojis.
-	// It's 1 character per emoji, plus 3 zero-width joiner characters between them.
-	ASSERT(length("👨‍👩‍👧‍👦") == 11)
-	ASSERT(length_char("👨‍👩‍👧‍👦") == 7)
+//Some setup code stuff
+/turf
+	var/isBlue = FALSE
+
+/turf/blue
+	isBlue = TRUE
+
+//The actual tests
+//NOTE: Tests placed in the IntegrationTests suite
+// should actually require a normal server in order to work.
+// Tests that only check generic DM behaviour that is not icon nor map-facing
+// should be done within Content.Tests.
+// ALSO NOTE: You do need to rebuild even if you just edit this document. Don't ask me why.
```

Comment:
The DMStandard folder in the DMCompiler project used to have this issue. It was fixed with eab44f650f104fee82274307e730df19af4383f6, maybe see if that works here too?